# Assignment #03
1. Explore College Scorecard website - https://collegescorecard.ed.gov/ 

2. Explore College Scorecard Data web page - https://collegescorecard.ed.gov/data/

3. Download and skim the documentation - https://collegescorecard.ed.gov/assets/FullDataDocumentation.pdf

4. Download the dataset - https://ed-public-download.app.cloud.gov/downloads/CollegeScorecard_Raw_Data.zip (265MB Zip File)

5. Unzip the file into a folder (there are many files, one for each school year)

6. Import the 2017-2018 school year file "MERGED2017_18_PP.csv" and explore:   

    * how many colleges or rows?   

    * how many variables or columns?   

    * Look at the first 20 rows   

    * look at the last 20 rows   

    * Look at a random sample of 20 rows    

7. Since there are too many variables to explore, many of which are not interesting. Modify your import statement to only import the following 9 variables:

- UNITID
- INSTNM
- CITY
- STABBR
- ZIP
- ADM_RATE
- UGDS
- TUITIONFEE_IN
- MN_EARN_WNE_P10 

8. Download and read the data dictionary file - https://collegescorecard.ed.gov/assets/CollegeScorecardDataDictionary.xlsx

Find out the definition of the above 9 variables and document the definitions in the Jupyter Notebook so that the readers will not need to repeat what you have done to know the meaning of these variables.

9. Use the predefined cells below to conduct and document your data exploration.

When in doubt, Google. Find examples and read documents/tutorials. 

Have fun!

In [1]:
# Import libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import random
import warnings
warnings.filterwarnings("ignore")

In [2]:
# load the 2017-2018 school year file "MERGED2017_18_PP.csv"   
data=pd.read_csv('CollegeScorecard_Raw_Data/MERGED2017_18_PP.csv')

In [3]:
# Find out how many colleges or rows?   
# Find out how many variables or columns? 
data.shape

(7112, 1977)

Data has 7112 rows and 1977 columns

In [4]:
len(data['INSTNM'].unique())

6978

There are 6978 Unique college names, but they might be from different region
For Example:
University of Maryland, Baltimore County
University of Maryland, Baltimore County (Shady Groove)

In [5]:
# Select all duplicate rows based on one column
# Tried to single out duplicates, but did not work properly. Will look into this again
duplicateRowsDF = data[data.duplicated(['INSTNM', 'CITY','STABBR'])]
len(duplicateRowsDF)
#print("Duplicate Rows based on a single column are:", duplicateRowsDF, sep='\n')

18

In [7]:
duplicateRowsDF.head(3)

UNITID    OPEID  OPEID6                                  INSTNM  \
919   143491  2349500   23495  Cannella School of Hair Design-Chicago   
3439  224679  2098300   20983               Western Technical College   
4634  421665  3109700   31097       Cadillac Institute of Cosmetology   

          CITY STABBR         ZIP  \
919    Chicago     IL  60632-2549   
3439   El Paso     TX       79927   
4634  Cadillac     MI  49601-1800   

                                           ACCREDAGENCY  \
919   National Accrediting Commission of Career Arts...   
3439  Accrediting Commission of Career Schools and C...   
4634                                                NaN   

                        INSTURL  \
919   cannellabeautyschools.com   
3439        www.westerntech.edu   
4634         instituteofcos.com   

                                              NPCURL  ...  \
919   cannellabeautyschools.com/cbs_disclosures.html  ...   
3439                   www.westerntech.edu/NetPrice/  ...   
4634          instituteofcos.net/netprice/npcalc.htm  ...   

      OMAWDP8_NOTFIRSTTIME_POOLED_SUPP  OMENRUP_NOTFIRSTTIME_POOLED_SUPP  \
919                                NaN                               NaN   
3439                            0.8107                            0.1183   
4634                               NaN                               NaN   

      OMENRYP_FULLTIME_POOLED_SUPP  OMENRAP_FULLTIME_POOLED_SUPP  \
919                            NaN                           NaN   
3439                             0                        0.0468   
4634                           NaN                           NaN   

      OMAWDP8_FULLTIME_POOLED_SUPP  OMENRUP_FULLTIME_POOLED_SUPP  \
919                            NaN                           NaN   
3439                        0.7857                        0.1676   
4634                           NaN                           NaN   

      OMENRYP_PARTTIME_POOLED_SUPP  OMENRAP_PARTTIME_POOLED_SUPP  \
919                            NaN                           NaN   
3439                             0                        0.0192   
4634                           NaN                           NaN   

      OMAWDP8_PARTTIME_POOLED_SUPP  OMENRUP_PARTTIME_POOLED_SUPP  
919                            NaN                           NaN  
3439                        0.6782                        0.3027  
4634                           NaN                           NaN  

[3 rows x 1977 columns]

In [8]:
# Look at the first 20 rows  
data.head(20)

UNITID    OPEID  OPEID6                                     INSTNM  \
0   100654   100200    1002                   Alabama A & M University   
1   100663   105200    1052        University of Alabama at Birmingham   
2   100690  2503400   25034                         Amridge University   
3   100706   105500    1055        University of Alabama in Huntsville   
4   100724   100500    1005                   Alabama State University   
5   100751   105100    1051                  The University of Alabama   
6   100760   100700    1007          Central Alabama Community College   
7   100812   100800    1008                    Athens State University   
8   100830   831000    8310            Auburn University at Montgomery   
9   100858   100900    1009                          Auburn University   
10  100937   101200    1012                Birmingham Southern College   
11  101028  1218200   12182     Chattahoochee Valley Community College   
12  101073  1055400   10554                  Concordia College Alabama   
13  101116  1303906   13039                South University-Montgomery   
14  101143   101500    1015         Enterprise State Community College   
15  101161   106000    1060          Coastal Alabama Community College   
16  101189   100300    1003                        Faulkner University   
17  101240   101700    1017            Gadsden State Community College   
18  101277  4187200   41872       New Beginning College of Cosmetology   
19  101286   101800    1018  George C Wallace Community College-Dothan   

              CITY STABBR         ZIP  \
0           Normal     AL       35762   
1       Birmingham     AL  35294-0110   
2       Montgomery     AL  36117-3553   
3       Huntsville     AL       35899   
4       Montgomery     AL  36104-0271   
5       Tuscaloosa     AL  35487-0166   
6   Alexander City     AL       35010   
7           Athens     AL       35611   
8       Montgomery     AL  36117-3596   
9           Auburn     AL       36849   
10      Birmingham     AL       35254   
11     Phenix City     AL       36869   
12           Selma     AL       36701   
13      Montgomery     AL       36116   
14      Enterprise     AL  36330-1300   
15     Bay Minette     AL  36507-2698   
16      Montgomery     AL  36109-3390   
17         Gadsden     AL       35903   
18     Albertville     AL       35951   
19          Dothan     AL  36303-9234   

                                         ACCREDAGENCY  \
0   Southern Association of Colleges and Schools C...   
1   Southern Association of Colleges and Schools C...   
2   Southern Association of Colleges and Schools C...   
3   Southern Association of Colleges and Schools C...   
4   Southern Association of Colleges and Schools C...   
5   Southern Association of Colleges and Schools C...   
6   Southern Association of Colleges and Schools C...   
7   Southern Association of Colleges and Schools C...   
8   Southern Association of Colleges and Schools C...   
9   Southern Association of Colleges and Schools C...   
10  Southern Association of Colleges and Schools C...   
11  Southern Association of Colleges and Schools C...   
12                                                NaN   
13  Southern Association of Colleges and Schools C...   
14  Southern Association of Colleges and Schools C...   
15  Southern Association of Colleges and Schools C...   
16  Southern Association of Colleges and Schools C...   
17  Southern Association of Colleges and Schools C...   
18  National Accrediting Commission of Career Arts...   
19  Southern Association of Colleges and Schools C...   

                                              INSTURL  \
0                                       www.aamu.edu/   
1                                         www.uab.edu   
2                           www.amridgeuniversity.edu   
3                                         www.uah.edu   
4                                       www.alasu.edu   
5                                         www.u

In [9]:
# Look at the last 20 rows 
data.tail(20)

UNITID    OPEID  OPEID6  \
7092  47503311  4208611   42086   
7093  47503312  4208612   42086   
7094  47503313  4208613   42086   
7095  47647001  4088301   40883   
7096  47662901  4189101   41891   
7097  47691101  4205801   42058   
7098  48065701   869423    8694   
7099  48079001  4218901   42189   
7100  48092101  4220001   42200   
7101  48133101  4213501   42135   
7102  48154401  4220901   42209   
7103  48305401  4224601   42246   
7104  48312401   108131    1081   
7105  48312402   108161    1081   
7106  48312403   108162    1081   
7107  48312404   108163    1081   
7108  48387801  4223701   42237   
7109  48400201  4228101   42281   
7110  48778201  4221501   42215   
7111  49005401  4182601   41826   

                                                 INSTNM              CITY  \
7092   Relay Graduate School of Education - San Antonio       San Antonio   
7093  Relay Graduate School of Education - Dallas-Fo...            Dallas   
7094  Relay Graduate School of Education - Washingto...        Washington   
7095                           WestMed College - Merced            Merced   
7096                  Salon Institute - Columbus Campus      Lewis Center   
7097          SAE Institute of Technology San Francisco        Emeryville   
7098                  Rasmussen College - Overland Park     Overland Park   
7099             Rocky Vista University - Southern Utah             Ivins   
7100                 Designer Barber & Stylist School -        Springdale   
7101             Sharp Edgez Barber Institute - Buffalo           Buffalo   
7102  National Personal Training Institute of Cleveland  Highland Heights   
7103             Barber School of Pittsburgh - Ambridge          Ambridge   
7104                 Arizona State University at Tucson            Tucson   
7105  Arizona State University - ASU Colleges at Lak...  Lake Havasu City   
7106        Arizona State University at The Gila Valley          Thatcher   
7107                   Arizona State University at Yuma              Yuma   
7108  Bay Area Medical Academy - San Jose Satellite ...          San Jose   
7109                        High Desert Medical College         Lancaster   
7110                                       BCI - Malden            Malden   
7111          Health Career Institute- Lauderdale Lakes  Lauderdale Lakes   

     STABBR        ZIP                                       ACCREDAGENCY  \
7092     TX  782155155       Middle States Commission on Higher Education   
7093     TX  752075155       Middle States Commission on Higher Education   
7094     DC  200375155       Middle States Commission on Higher Education   
7095     CA  953400000                                                NaN   
7096     OH  430359153  National Accrediting Commission of Career Arts...   
7097     CA      94608                                                NaN   
7098     KS  662102786                         Higher Learning Commission   
7099     UT  847386756                         Higher Learning Commission   
7100     AR  727620990  National Accrediting Commission of Career Arts...   
7101     NY  142170000                  Council on Occupational Education   
7102     OH      44143  Accrediting Council for Continuing Education &...   
7103     PA  150030000                                                NaN   
7104     AZ  857454284                         Higher Learning Commission   
7105     AZ  864036877                         Higher Learning Commission   
7106     AZ  855525545                         Higher Learning Commission   
7107     AZ  853656900                         Higher Learning Commission   
7108     CA      95113  Accrediting Council for Independent Colleges a...   
7109     CA  935340000  Accrediting Council for Continuing Education &...   
7110     MA  021480000                                                NaN   
7111     FL  333137268  Accrediting Commission of Career Schools and C...   

                                    

In [10]:
# Look at a random sample of 20 rows    
data.sample(20)

UNITID    OPEID  OPEID6  \
6762  15098703   991713    9917   
5259    449117   499201    4992   
2033    182564  1036300   10363   
1185    153621   187100    1871   
649     131496   144500    1445   
254     111799  2332813   23328   
1860    176479   244700    2447   
3454    225432   361200    3612   
747     136765  2511700   25117   
1557    166027   215500    2155   
4517    417406  3090600   30906   
1883    177278   246200    2462   
4914    441238  3592400   35924   
5687    458885   145984    1459   
5311    450021  3294301   32943   
5232    448576  2179948   21799   
6584    489937   295600    2956   
6036    476869  4203500   42035   
2985    212054   325600    3256   
687     133881   146900    1469   

                                                 INSTNM            CITY  \
6762            Ivy Tech Community College-East Central          Muncie   
5259                    Platt College-Miller-Motte-Cary            Cary   
2033                             Western Nevada College     Carson City   
1185                           Iowa Wesleyan University  Mount Pleasant   
649                               Georgetown University      Washington   
254                                         CET-Salinas         Salinas   
1860                           William Carey University     Hattiesburg   
3454                     University of Houston-Downtown         Houston   
747                             Ridge Technical College    Winter Haven   
1557                                 Harvard University       Cambridge   
4517                            Cobb Beauty College Inc        Kennesaw   
1883                          Eden Theological Seminary  Webster Groves   
4914  Irell & Manella Graduate School of Biological ...          Duarte   
5687                        Strayer University-Arkansas     Little Rock   
5311                           Blue Cliff College-Houma           Houma   
5232  Argosy University-The Art Institute of Califor...  San Bernardino   
6584                  Piedmont International University   Winston-Salem   
6036                                Austin Kade Academy     Idaho Falls   
2985                                  Drexel University    Philadelphia   
687                     Florida Institute of Technology       Melbourne   

     STABBR         ZIP                                       ACCREDAGENCY  \
6762     IN   473029448                         Higher Learning Commission   
5259     NC  27518-7171                                                NaN   
2033     NV  89703-7399  Northwest Commission on Colleges and Universities   
1185     IA       52641                         Higher Learning Commission   
649      DC  20057-0001       Middle States Commission on Higher Education   
254      CA       93906                  Council on Occupational Education   
1860     MS  39401-5499  Southern Association of Colleges and Schools C...   
3454     TX  77002-1014  Southern Association of Colleges and Schools C...   
747      FL  33881-9518                  Council on Occupational Education   
1557     MA       02138         New England Commission on Higher Education   
4517     GA  30144-2828                                                NaN   
1883     MO  63119-3192                         Higher Learning Commission   
4914     CA  91010-3000  Western Association of Schools and Colleges Se...   
5687     AR       72211       Middle States Commission on Higher Education   
5311     LA       70360  Accrediting Commission of Career Schools and C...   
5232     CA  92408-2800                                                NaN   
6584     NC       27101  Transnational Association of Christian College...   
6036     ID  83404-5540  National Accrediting Commission of Career Arts...   
2985     PA       19104       Middle States Commission on Higher Education   
687      FL  32901-6975  Southern Association of Colleges and Schools C...   

                                     INSTURL  \
6762     

Read (with naked eyes, not computer) the dictionary and find out the definitions of the following 12 columns and copy the text next to the column name
- UNITID: Unit ID for institution           
- INSTNM:  Institution name        
- CITY: City in which University is located 
- STABBR: State Abbrevation
- ZIP: Zip Code of the location of the University             
- ADM_RATE: Admission Rate in a particular college 
- UGDS: Undergrads Count  
- TUITIONFEE_IN: In Tuition Fee for a particular college  
- MN_EARN_WNE_P10: Mean earnings of students working and not enrolled 10 years after entry       

In [11]:
# reimport the data with above 12 columns (ignore all other variables)
data=pd.read_csv('CollegeScorecard_Raw_Data/MERGED2017_18_PP.csv', usecols = ['UNITID','INSTNM','CITY','STABBR','ZIP',
                                                                              'ADM_RATE','UGDS','TUITIONFEE_IN','MN_EARN_WNE_P10'])
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7112 entries, 0 to 7111
Data columns (total 9 columns):
UNITID             7112 non-null int64
INSTNM             7112 non-null object
CITY               7112 non-null object
STABBR             7112 non-null object
ZIP                7112 non-null object
ADM_RATE           2037 non-null float64
UGDS               6364 non-null float64
TUITIONFEE_IN      4021 non-null float64
MN_EARN_WNE_P10    0 non-null float64
dtypes: float64(4), int64(1), object(4)
memory usage: 500.2+ KB


In [12]:
data.head(5)

UNITID                               INSTNM        CITY STABBR         ZIP  \
0  100654             Alabama A & M University      Normal     AL       35762   
1  100663  University of Alabama at Birmingham  Birmingham     AL  35294-0110   
2  100690                   Amridge University  Montgomery     AL  36117-3553   
3  100706  University of Alabama in Huntsville  Huntsville     AL       35899   
4  100724             Alabama State University  Montgomery     AL  36104-0271   

   ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
0    0.9027   4824.0         9857.0              NaN  
1    0.9181  12866.0         8328.0              NaN  
2       NaN    322.0         6900.0              NaN  
3    0.8123   6917.0        10280.0              NaN  
4    0.9787   4189.0        11068.0              NaN

In [18]:
# Find out the 10 colleges that are hardest to get in (top 10)
## This can be found using the admission rate. The lowest admission rate of any
## college suggests that it is tougher to get in.
toughest_colleges=data.nsmallest(10, ['ADM_RATE']) 
toughest_colleges.head(10)

UNITID                                             INSTNM  \
2974  211635                         Citizens School of Nursing   
5226  448309  Shorter University-College of Adult & Professi...   
6112  480912       Gemini School of Visual Arts & Communication   
6664  491288             University of Wisconsin-Milwaukee Flex   
2978  211893                          Curtis Institute of Music   
3979  243744                                Stanford University   
1557  166027                                 Harvard University   
2302  192110                               The Juilliard School   
2123  186131                               Princeton University   
2233  190150        Columbia University in the City of New York   

                CITY STABBR         ZIP  ADM_RATE    UGDS  TUITIONFEE_IN  \
2974  New Kensington     PA  15068-6525    0.0000   158.0            NaN   
5226            Rome     GA       30165    0.0000   113.0         9680.0   
6112      Cedar Park     TX       78613    0.0000    18.0            NaN   
6664       Milwaukee     WI       53201    0.0000   501.0         6750.0   
2978    Philadelphia     PA       19103    0.0330   131.0         2675.0   
3979        Stanford     CA       94305    0.0473  7056.0        49617.0   
1557       Cambridge     MA       02138    0.0516  7532.0        48949.0   
2302        New York     NY  10023-6588    0.0592   486.0        43170.0   
2123       Princeton     NJ  08544-0070    0.0641  5246.0        47140.0   
2233        New York     NY       10027    0.0661  8170.0        57208.0   

      MN_EARN_WNE_P10  
2974              NaN  
5226              NaN  
6112              NaN  
6664              NaN  
2978              NaN  
3979              NaN  
1557              NaN  
2302              NaN  
2123              NaN  
2233              NaN

In [20]:
# Below are the names of the colleges
toughest_colleges['INSTNM']

2974                           Citizens School of Nursing
5226    Shorter University-College of Adult & Professi...
6112         Gemini School of Visual Arts & Communication
6664               University of Wisconsin-Milwaukee Flex
2978                            Curtis Institute of Music
3979                                  Stanford University
1557                                   Harvard University
2302                                 The Juilliard School
2123                                 Princeton University
2233          Columbia University in the City of New York
Name: INSTNM, dtype: object

In [26]:
# 10. Find out the average in-state tuition of top 10
#toughest_colleges.mean()
toughest_colleges.loc[:,'TUITIONFEE_IN'].mean()

33148.625

In [27]:
# Find out the average earnings of graduates from the top 10
# If we look at the data, all the entires have a null vale (missing data)
print('Average earnings of graduates from the top 10: ')
toughest_colleges.loc[:,'MN_EARN_WNE_P10'].mean()

nan

In [28]:
# Find out the 10 colleges that are easiest to get in (bottom 10)
# Easiest colleges mean that the admission rate has to be higher

easiest_colleges=data.nlargest(10, ['ADM_RATE']) 
easiest_colleges.head(10)

UNITID                                      INSTNM        CITY STABBR  \
24    101453               Heritage Christian University    Florence     AL   
240   110918                California Christian College      Fresno     CA   
370   118161    Lytles Redwood Empire Beauty College Inc  Santa Rosa     CA   
472   123280    Shasta Bible College and Graduate School     Redding     CA   
559   127653                           Naropa University     Boulder     CO   
571   127954  Montessori Education Center of the Rockies     Boulder     CO   
652   131830      National Conservatory of Dramatic Arts  Washington     DC   
665   132842              Carlos Albizu University-Miami       Miami     FL   
887   142090                         Boise Bible College       Boise     ID   
1011  147165                            Midstate College      Peoria     IL   

             ZIP  ADM_RATE   UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
24    35630-9977       1.0   59.0        11232.0              NaN  
240        93727       1.0   16.0         9330.0              NaN  
370        95403       1.0  158.0            NaN              NaN  
472        96002       1.0   42.0        12060.0              NaN  
559   80302-6697       1.0  384.0        31790.0              NaN  
571        80301       1.0   16.0            NaN              NaN  
652        20007       1.0   32.0         9900.0              NaN  
665   33172-2209       1.0  236.0        12444.0              NaN  
887   83714-1220       1.0  128.0        12080.0              NaN  
1011  61614-3595       1.0  274.0        16230.0              NaN

In [31]:
# Find out the average in-state tuition of the bottom 10
print('Average in-state tuition fee of the bottom 10: ')
easiest_colleges.loc[:,'TUITIONFEE_IN'].mean()

Average in-state tuition fee of the bottom 10


14383.25

In [32]:
# Find out the average earnings of graduates from the bottom 10
# If we look at the data, all the entires have a null vale (missing data)
print('Average earnings of graduates from the bottom 10: ')
easiest_colleges.loc[:,'MN_EARN_WNE_P10'].mean()

Average earnings of graduates from the bottom 10: 


nan